packages/core/src/RenderingEngine/VideoViewport.ts

```diff
@@ -0,0 +1,509 @@
+    const seekEventListener = (evt) => {
+      renderFrame();
+
+      videoElement.removeEventListener('seeked', seekEventListener);
+    };
+
+    videoElement.addEventListener('seeked', seekEventListener);
+  }
+
+  public async start() {
+    const videoElement = this.videoElement;
+    const renderFrame = this.renderFrame;
+
+    videoElement.currentTime = 0;
+
+    if (videoElement.paused) {
+      // Need to wait for seek update
+      const seekEventListener = (evt) => {
+        console.log('seeked');
+
+        renderFrame();
+
+        videoElement.removeEventListener('seeked', seekEventListener);
+      };
+
+      videoElement.addEventListener('seeked', seekEventListener);
+    }
+  }
+
+  public async end() {
+    const videoElement = this.videoElement;
+    const renderFrame = this.renderFrame;
+
+    videoElement.currentTime = videoElement.duration;
+
+    if (videoElement.paused) {
+      // Need to wait for seek update
+      const seekEventListener = (evt) => {
+        renderFrame();
+
+        videoElement.removeEventListener('seeked', seekEventListener);
+      };
+
+      videoElement.addEventListener('seeked', seekEventListener);
+    }
+  }
+
+  public async setTime(timeInSeconds: number) {
+    const videoElement = this.videoElement;
+    const renderFrame = this.renderFrame;
+
+    videoElement.currentTime = timeInSeconds;
+
+    if (videoElement.paused) {
+      // Need to wait for seek update
+      const seekEventListener = (evt) => {
+        renderFrame();
+
+        videoElement.removeEventListener('seeked', seekEventListener);
+      };
+
+      videoElement.addEventListener('seeked', seekEventListener);
+    }
+  }
+
+  // Sets the frame number - note according to DICOM, this is 1 based
+  public async setFrame(frame: number) {
+    this.setTime((frame - 1) / this.fps);
+  }
+
+  public setProperties(videoInterface: VideoViewportProperties) {
+    if (videoInterface.loop !== undefined) {
+      this.videoElement.loop = videoInterface.loop;
+    }
+
+    if (videoInterface.muted !== undefined) {
+      this.videoElement.muted = videoInterface.muted;
+    }
+
+    if (videoInterface.playbackRate !== undefined) {
+      this.setPlaybackRate(videoInterface.playbackRate);
+    }
+  }
+
+  public setPlaybackRate(rate = 1) {
+    // Minimum playback speed in chrome is 0.0625 compared to normal
+    if (rate < 0.0625) {
+      this.pause();
+      return;
+    }
+    if (!this.videoElement) {
+      return;
+    }
+    this.videoElement.playbackRate = rate;
+    this.play();
+  }
+
+  public setScrollSpeed(scrollSpeed = 1, unit = 'f') {
+    this.scrollSpeed = unit === 's' ? scrollSpeed * this.fps : scrollSpeed;
+  }
+
+  public getProperties = (): VideoViewportProperties => {
+    return {
+      loop: this.videoElement.loop,
+      muted: this.videoElement.muted,
+    };
+  };
+
+  public resetProperties() {
+    this.setProperties({
+      loop: false,
+      muted: true,
+    });
+  }
+
+  public setCamera(
```

Comment:
same for this, I'm already seeing some custom logic for the pan and zoom in the tools. And it will happen for other tools. 

As mentioned there is no mention of CPU vs GPU in the Pan/Zoom tool since we take care of them at the viewport level and their camera are two different cameras. But we force them to implement the same interface ICamera. 

Then when setCameraCPU we just convert them back to internal CPU camera